src/map_io/map_buildingdata_packet.cc

```diff
@@ -744,10 +744,20 @@ void MapBuildingdataPacket::read_productionsite(
 				}
 			}
 
-			uint16_t const stats_size = fr.unsigned_16();
-			productionsite.statistics_.resize(stats_size);
-			for (uint32_t i = 0; i < productionsite.statistics_.size(); ++i)
-				productionsite.statistics_[i] = fr.unsigned_8();
+			// TODO(hessenfarmer): Savegame compatibility, remove after Build 21.
+			if (packet_version >= 8) {
+				productionsite.crude_percent_ = fr.unsigned_32();
+			} else {
+				uint16_t const stats_size = fr.unsigned_16();
+				uint8_t ok = 0;
+				for (uint32_t i = 0; i < stats_size; ++i) {
+					if (fr.unsigned_8()) {
+						ok++;
+					}
+				}
```

Comment:
Suggestion for needing one variable less:
```
uint16_t ok = 0;
for (uint16_t i = fr.unsigned_16(); i; --i) {
  if (fr.unsigned_8)) {
    ++ok;
  }
}
```

Reply:
okay will try this

Reply:
doesn't work as I need to divide by the length of the old stats vector